### ISSUE TITLE
buildah run fails when using --runtime=kata-runtime 

### ISSUE THREAD

**Issue by stephen-dexda** [containers/buildah#2524]
**Description**

Running any command with `buildah run --runtime=kata-runtime ...` results in an error (see example below).

Running Podman with `--runtime=kata-runtime` (`create`, `run` etc.) works fine.

Using Kata when executing less-trusted binaries via `buildah run` is desirable to take advantage of the reduced attack surface of virtualisation vs. containerisation.

**Steps to reproduce the issue:**

1. `container="$(sudo buildah from registry.fedoraproject.org/fedora)"`
2. `sudo buildah run --runtime=kata-runtime "$container" ls`

Commands are run with `sudo` since Kata can't run rootless [yet](https://github.com/kata-containers/runtime/issues/2132.

**Describe the results you received:**

```
rpc error: code = Internal desc = Could not run process: container_linux.go:349: starting container process caused "exec: \"/usr/bin/ls\": stat /usr/bin/ls: no such file or directory"
error running container: error creating container for [/usr/bin/ls]: : exit status 1
error while running runtime: exit status 1
```

**Describe the results you expected:**

Listing of `/`.

**Output of `rpm -q buildah` or `apt list buildah`:**

```
buildah-1.15.1-1.fc32.x86_64
```

**Comment by rhatdan** [comment]
Could this be an SELinux issue?

**Comment by stephen-dexda** [comment]
Just tested after `setenforce 0`, same problem.

**Comment by rhatdan** [comment]
@giuseppe PTAL

**Comment by giuseppe** [comment]
can you please attach the output when you run with `buildah --log-level=debug` ?

**Comment by stephen-dexda** [comment]
[buildah-kata.log](https://github.com/containers/buildah/files/5058532/buildah-kata.log)

**Comment by stephen-dexda** [comment]
Also, kata-runtime messages from journald (from running the above yesterday):

[kata-runtime.log](https://github.com/containers/buildah/files/5058575/kata-runtime.log)

**Comment by giuseppe** [comment]
I think this happens because buildah sets up an intermediate mount namespace, and the created mount point is not accessible from the host.

I think we could to drop the login of creating an intermediate namespace and do what we do in Podman.

https://github.com/containers/buildah/pull/2543 seems to solve the issue for now.

**Comment by giuseppe** [comment]
unfortunately the fix I had in #2543 is not enough.  There are other issues with kata, we cannot wait for the the PID returned in the PIDfile.

I suggest using podman for using kata:

```
container="$(sudo buildah from registry.fedoraproject.org/fedora)"
mnt=$(buildah mount $container)
podman --runtime kata-runtime run --rm -ti --rootfs $mnt ls /
```

and probably that is what we should look at as a solution in the long run instead of handling containers in buildah.

**Comment by stephen-dexda** [comment]
That workaround's very helpful, thanks Giuseppe.

I slightly modified it since all the commands need `sudo` (assuming current login is non-root) and to unmount the root FS afterwards for consistency with state after the equivalent Buildah run command:

```bash
kata_buildah_run() {
    local container="$1"
    local args=("${@:2}")
    local mnt_dir
    mnt_dir="$(sudo buildah mount "$container")"
    sudo podman --runtime=/usr/bin/kata-runtime run --rm -it --rootfs "$mnt_dir" "${args[@]}"
    sudo buildah unmount "$container"
}
```
